src/YEthStakerStrategy.sol

```diff
@@ -0,0 +1,404 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.18;
+
+import {BaseStrategy, ERC20} from "@tokenized-strategy/BaseStrategy.sol";
+import {CustomStrategyTriggerBase} from "@periphery/ReportTrigger/CustomStrategyTriggerBase.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ICurvePool} from "./interfaces/ICurvePool.sol";
+import {IYEthStaker} from "./interfaces/IYEthStaker.sol";
+import {IYEthPool} from "./interfaces/IYEthPool.sol";
+import {IDepositFacility} from "./interfaces/IDepositFacility.sol";
+import {ICommonReportTrigger} from "./interfaces/ICommonReportTrigger.sol";
+
+/**
+ * The `TokenizedStrategy` variable can be used to retrieve the strategies
+ * specific storage data your contract.
+ *
+ *       i.e. uint256 totalAssets = TokenizedStrategy.totalAssets()
+ *
+ * This can not be used for write functions. Any TokenizedStrategy
+ * variables that need to be updated post deployment will need to
+ * come from an external call from the strategies specific `management`.
+ */
+
+// NOTE: To implement permissioned functions you can use the onlyManagement, onlyEmergencyAuthorized and onlyKeepers modifiers
+
+contract YEthStakerStrategy is BaseStrategy, CustomStrategyTriggerBase {
+    using SafeERC20 for ERC20;
+
+    ICurvePool public constant curvepool =
+        ICurvePool(0x69ACcb968B19a53790f43e57558F5E443A91aF22); // 0 is WETH, 1 is yETH
+    ERC20 public constant WETH =
+        ERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    ERC20 public constant yETH =
+        ERC20(0x1BED97CBC3c24A4fb5C069C6E311a967386131f7);
+    IYEthStaker public constant styETH =
+        IYEthStaker(0x583019fF0f430721aDa9cfb4fac8F06cA104d0B4);
+    IYEthPool public constant yETHPool =
+        IYEthPool(0x2cced4ffA804ADbe1269cDFc22D7904471aBdE63);
+    ICommonReportTrigger public constant COMMON_REPORT_TRIGGER =
+        ICommonReportTrigger(0xD98C652f02E7B987e0C258a43BCa9999DF5078cF);
+
+    int128 internal constant WETH_INDEX = 0;
+    int128 internal constant YETH_INDEX = 1;
+    uint256 internal constant MAX_BPS = 10000;
+    uint256 internal constant WAD = 1e18;
+
+    address immutable GOV;
+
+    IDepositFacility public depositFacility;
+    uint256 public maxSingleWithdraw = 100 * 1e18;
+    uint256 public swapSlippage = 50;
+
+    event DepositFacilitySet(address facility);
+    event MaxSingleWithdrawSet(uint256 max);
+    event SwapSlippageSet(uint256 slippage);
+
+    constructor(
+        address _asset,
+        string memory _name,
+        address _gov
+    ) BaseStrategy(_asset, _name) {
+        require(_asset == address(WETH), "Asset!=WETH");
+        WETH.approve(address(curvepool), type(uint256).max);
+        yETH.approve(address(curvepool), type(uint256).max);
+        yETH.approve(address(styETH), type(uint256).max);
+        require(_gov != address(0), "GOV=0x0");
```

Comment:
instead of gov let's just use the roles that are in v3: role_manager

Reply:
Please check: https://github.com/yearn/yearn-vaults-v3/blob/master/contracts/interfaces/Roles.sol

EmergencyManager